test/parallel/test-quic-updatekey.js

```diff
@@ -0,0 +1,70 @@
+// Flags: --no-warnings
+'use strict';
+
+// Test QuicSocket constructor option errors.
+
+const common = require('../common');
+if (!common.hasQuic)
+  common.skip('missing quic');
+
+const assert = require('assert');
+const Countdown = require('../common/countdown');
+const fixtures = require('../common/fixtures');
+const key = fixtures.readKey('agent1-key.pem', 'binary');
+const cert = fixtures.readKey('agent1-cert.pem', 'binary');
+const ca = fixtures.readKey('ca1-cert.pem', 'binary');
+
+const { createSocket } = require('quic');
+const kServerPort = process.env.NODE_DEBUG_KEYLOG ? 5678 : 0;
+const kClientPort = process.env.NODE_DEBUG_KEYLOG ? 5679 : 0;
+let client;
+
+const server = createSocket({ port: kServerPort, validateAddress: true });
+server.setDiagnosticPacketLoss({ rx: 0.00, tx: 0.00 });
+const kALPN = 'meow';  // ALPN can be overriden to whatever we want
+const kServerName = 'agent2';  // Intentionally the wrong servername
+
+
+const countdown = new Countdown(2, () => {
+  server.close();
+  client.close();
+});
+
+server.listen({
+  key,
+  cert,
+  ca,
+  requestCert: true,
+  rejectUnauthorized: false,
+  alpn: kALPN,
+  maxCryptoBuffer: 4096,
+});
+server.on('session', common.mustCall((session) => {
+  process.nextTick(() => {
+    assert.throws(() => session.updateKey(), {
```

Comment:
So, investigating further... the issue around updateKey is that a key update can only be triggered (a) only after the TLS handshake has been confirmed and (b) while we're not in the middle of another key update. The TLS handshake is confirmed as soon as we receive the first 1RTT bytes from the peer but ngtcp2 does not really give us a notification at that point so we're going to have to guess somewhat. I'll be working through the semantics on that and will be implementing a `'handshakeConfirmed'` type of event that will flip the switch that enables `updateKey()` to work. Expect that within the next week I hope

Reply:
Sounds good @jasnell, tell me if you need some help.